src/ossl/sshkdf.rs

```diff
@@ -137,7 +141,15 @@ impl Derive for SSHKDFOperation {
             return Err(CKR_DEVICE_ERROR)?;
         }
 
-        self.fips_approved = check_kdf_fips_indicators(&mut kctx)?;
+        #[cfg(feature = "fips")]
+        {
+            if ossl_fips_indicator_is_set() {
+                self.fips_approved = Some(false);
+            }
+            if self.fips_approved.is_none() {
+                self.fips_approved = Some(true);
+            }
+        }
```

Comment:
This is repeated in several places so reviewing all the cases do the right thing is pain. Given that the `fips_approved` is set on different objects, making a function from that won't easily work (?) so should we go ahead with some macro?

Reply:
macros would be worse imo.
But we could create a function that takes &mut arguments in theory ...
Do we need to centralize this?